### DISCUSSION TITLE
Can we have a new "<<qualify-widgettree>>" macro?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5943] [Ideas]
I would suggest adding a macro that adds the `footprint` in the widget-tree to the state qualifier like I've already done before

We could call it `<<qualify-widgettree>>` or whatever

It would help in generating states that would otherwise be the same using the standard `<<qualify>>` macro

**Comment by pmario** [comment]
I would like it to be a shorter name eg: `<<qualify-tree>>`  or `<<qualify-fp>>` or something similar

**Reply by BurningTreeC** [reply]
Yes, the name doesn't matter to me. The question is if it's reasonable to add this as a standalone macro

**Comment by Jermolene** [comment]
Hi @BurningTreeC interesting idea. I don't think we should merge it into the core until it is being used by the core, but perhaps that depends on the complexity of the implementation and the value of the usecases. Worth exploring in any case. As for naming, I think I'd favour something like `<<current-widget-id>>`.
